### ISSUE TITLE
[BUG] npm dependencies seem not to be installed reproducibly

### ISSUE THREAD

**Issue by florianschieder** [django-cms/django-cms#7388]
## Description

We recently encountered unreproducible issues while editing a page within our CMS which is based on DjangoCMS 3.9.0.
I found out a possible reason: The npm dependencies seem not to be installed reproducibly.

[DjangoCMS package.json (3.9.0)](https://github.com/django-cms/django-cms/blob/release/3.9.x/package.json) defines some "compatible with"-dependencies (see [here](https://docs.npmjs.com/cli/v6/using-npm/semver#caret-ranges-123-025-004)).

Please take the dependency `"autoprefixer": "^6.3.6"` as defined within the `package.json` linked above as example:
1. If the most recently released version of `autoprefixer` is `6.3.6`, version `6.3.6` will be installed.
2. If the most recently released version of `autoprefixer` is `6.11.0`, version `6.11.0` will be installed.
3. If the most recently released version of `autoprefixer` is `7.0` and the most recent version within the `6.x.x` branch is `6.11.0`, version `6.11.0` will be installed. 
4. If the most recently released version of `autoprefixer` is `7.0` and the most recent version within the `6.x.x` branch is `6.11.1` (for example, a critical security bugfix), version `6.11.1` will be installed.

IMHO if I don't get anything wrong, this would cause that if the second scenario occured and version `6.11.0` contained a bug, projects depending on `django-cms` may encounter issues until `django-cms` rolled back to another `autoprefixer` version or until `autoprefixer` fixed their bugs.

I also noticed that the [DjangoCMS development branch](https://github.com/django-cms/django-cms/tree/develop/) now contains a `package-lock.json` to freeze the dependencies. IMHO this is not absolutely effective yet, because [install-npm-dependencies.sh](https://github.com/django-cms/django-cms/blob/develop/scripts/install-npm-dependencies.sh) uses [npm install](https://docs.npmjs.com/cli/v8/commands/npm-install) and not [npm clean-install](https://docs.npmjs.com/cli/v8/commands/npm-ci) which means that IMHO the installation is not entirely frozen.

## Do you want to help fix this issue?

* [ ] Yes, I want to help fix this issue and I will join #workgroup-pr-review on [Slack](https://www.django-cms.org/slack) to confirm with the community that a PR is welcome. 
* [x] No, I only want to report the issue.

**Comment by marksweb** [comment]
@florianschieder Thanks for reporting. What issue did you encounter?

I don't believe [install-npm-dependencies.sh](https://github.com/django-cms/django-cms/blob/develop/scripts/install-npm-dependencies.sh) has been used in 3.9, 3.10 or 3.11.

The release script uses `prepare-buildenv.sh`, although this still uses `npm install`, I'll admit. We're still trying to improve our build & release processes, so if there are changes that can be brought in then that's great.

**Comment by florianschieder** [comment]
Thanks very much for your quick response.

The problem we encountered was that after we deployed our CMS based on DjangoCMS at the 8th of september, saving a plugin led to a page reload. That has been very annoying and DjangoCMS never behaved like that before.

After I rebuilt all dependencies the day before yesterday everything was fine again. Since our dependency installations are absolutely frozen I concluded from this it must have been an external dependency issue.
